exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillOperatorTable.java

```diff
@@ -157,9 +159,23 @@ private void populateWrappedCalciteOperators() {
       } else if(calciteOperator instanceof SqlFunction) {
         wrapper = new DrillCalciteSqlFunctionWrapper((SqlFunction) calciteOperator,
             getFunctionListWithInference(calciteOperator.getName()));
+      } else if(calciteOperator instanceof SqlBetweenOperator) {
+        // During the procedure of converting to RexNode,
+        // StandardConvertletTable.convertBetween expects the SqlOperator to be a subclass of SqlBetweenOperator
+        final SqlBetweenOperator sqlBetweenOperator = (SqlBetweenOperator) calciteOperator;
+        wrapper = new SqlBetweenOperator(sqlBetweenOperator.flag, sqlBetweenOperator.isNegated()) {
+          @Override
+          public boolean checkOperandTypes(
```

Comment:
Does it make sense to fix this in Calcite?

It also seems like we need to have a better merging strategy between Drill operator table and calcite operator table. One of the problems we see is DATE - DATE doesn't work on schemed types.